crates/graphql/src/schema/query_root.rs

```diff
@@ -720,6 +721,40 @@ impl QueryRoot {
             .collect::<Vec<MetadataJson>>())
     }
 
+    #[graphql(
+        description = "Returns collection data along with collection activities",
+        arguments(address(description = "Collection address"),)
+    )]
+    async fn collection(
+        &self,
+        context: &AppContext,
+        address: String,
+    ) -> FieldResult<Option<Collection>> {
+        let conn = context.shared.db.get()?;
+
+        metadatas::table
+            .inner_join(
+                metadata_jsons::table.on(metadatas::address.eq(metadata_jsons::metadata_address)),
+            )
+            .inner_join(
+                metadata_collection_keys::table
+                    .on(metadata_collection_keys::collection_address.eq(metadatas::mint_address)),
+            )
+            .inner_join(
+                current_metadata_owners::table
+                    .on(current_metadata_owners::mint_address.eq(metadatas::mint_address)),
+            )
+            .filter(metadata_collection_keys::collection_address.eq(address))
+            .filter(metadata_collection_keys::verified.eq(true))
+            .select(queries::metadatas::NFT_COLUMNS)
+            .first::<models::Nft>(&conn)
+            .optional()
+            .context("Failed to load NFT by metadata address.")?
+            .map(TryInto::try_into)
+            .transpose()
+            .map_err(Into::into)
+    }
```

Comment:
Can you please bundle this query in the queries::collection module under show function? Keep the query root more readable that way.